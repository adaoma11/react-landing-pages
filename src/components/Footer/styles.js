import styled, { css } from 'styled-components';

export const Footer = styled.footer`
  ${({ theme, $hasBg }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    height: 150px;
    background-color: ${$hasBg ? theme.colors.primary : theme.colors.white};
  `}
`;
