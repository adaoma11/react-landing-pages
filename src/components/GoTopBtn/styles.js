import styled, { css } from 'styled-components';

export const Link = styled.a`
  ${({ theme }) => css`
    display: flex;
    width: 40px;
    height: 40px;
    position: fixed;
    bottom: 2rem;
    right: 2rem;
    color: ${theme.colors.white};
    background-color: ${theme.colors.primary};
    border-radius: 5px;
    opacity: 0.8;
  `}
`;
