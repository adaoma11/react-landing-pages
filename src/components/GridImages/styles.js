import styled from 'styled-components';

export const GridContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  width: 100%;
  gap: 20px;
  margin-top: 50px;
`;

export const GridItem = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 0.5s ease-in-out;

  &:hover {
    transform: scale(1.2) rotate(5deg);
  }
`;
