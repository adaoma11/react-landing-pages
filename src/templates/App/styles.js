import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
  `}
`;
