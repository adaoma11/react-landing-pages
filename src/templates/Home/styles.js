import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
    background-color: ${theme.colors.primary};
  `}
`;
