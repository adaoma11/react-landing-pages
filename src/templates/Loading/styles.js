import styled, { css, keyframes } from 'styled-components';

const rotate = () => keyframes`
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
`;

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    width: 100%;
    height: 100vh;
    align-items: center;
    justify-content: center;
    background-color: ${theme.colors.primary};

    &:after {
      content: '';
      width: 50px;
      height: 50px;
      border-top: 5px solid ${theme.colors.white};
      border-right: 5px solid ${theme.colors.white};
      border-bottom: 5px solid transparent;
      border-left: 5px solid transparent;
      border-radius: 50%;
      animation: ${rotate()} 1s linear infinite;
    }
  `}
`;
