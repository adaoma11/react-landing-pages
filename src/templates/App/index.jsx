import * as Styled from "./styles";

function App() {
  return (
    <>
      <Styled.Wraper>
        <h1>Hello world</h1>
      </Styled.Wraper>
    </>
  );
}

export default App;
