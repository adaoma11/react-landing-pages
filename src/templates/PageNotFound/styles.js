import styled, { css } from 'styled-components';
import { P as Text } from '../../components/Text/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    width: 100%;
    height: 100vh;
    align-items: center;
    justify-content: center;
    background-color: ${theme.colors.primary};

    & > ${Text} > a {
      text-decoration: none;
      color: ${theme.colors.white};
    }

    & > ${Text} > a:hover {
      color: ${theme.colors.secondary};
    }
  `}
`;
