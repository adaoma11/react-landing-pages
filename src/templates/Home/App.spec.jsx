import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { theme } from '../../styles/theme';
import App from '.';

describe('App', () => {
  it("should render 'hello world'", () => {
    renderTheme(<App />);

    const headingContainer = screen.getByRole('heading', {
      name: /hello/i,
    }).parentElement;

    expect(headingContainer).toHaveStyle({ background: theme.colors.mainBg });
    expect(headingContainer).toMatchSnapshot();
    expect(headingContainer).toHaveStyleRule('justify-content', 'center');
  });
});
