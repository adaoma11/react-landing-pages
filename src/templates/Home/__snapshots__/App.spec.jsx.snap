// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 'hello world' 1`] = `
.c0 {
  color: #fff;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
}

<div
  class="c0"
>
  <h1>
    Hello world
  </h1>
</div>
`;
